### PR TITLE
remove locks in get service address for proxy

### PR DESCRIPTION
**Pull request by ramaraochavali**
This lock is very costly and has impact on push timing. An attempt to remove it.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -1038,6 +1042,9 @@ func (ps *PushContext) initServiceRegistry(env *Environment) error {
 		}
 		ps.ServiceByHostnameAndNamespace[s.Hostname][s.Attributes.Namespace] = s
 		ps.ServiceByHostname[s.Hostname] = s
+		s.Mutex.RLock()
+		ps.ClusterVIPs[s] = s.ClusterVIPs
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context.go:1046]: This can cause panic , the map is still that map

Comment (ramaraochavali) [reply pilot/pkg/model/push_context.go]: Ah. Thanks. Fixed

**Review by rshriram — APPROVED** [review]

**Pull request merged**